js/app/Visualization/UI/LayerReportDialog.js

```diff
@@ -0,0 +1,66 @@
+define([
+  "app/Logging",
+  "app/Visualization/UI/MessageDialog",
+  "dijit/Dialog",
+  "jQuery"
+], function(
+  Logging,
+  MessageDialog,
+  Dialog,
+  $
+) {
+  return {
+    show: function(visualizationState, layerName, layerData) {
+      var endTime = visualizationState.values.time;
+      var offset = visualizationState.values.timeExtent;
+      var startTime = new Date(endTime.getTime() + offset);
+
+      var message =
+        "Do you want to generate a vessel activity report for " +
+        layerName + " " + layerData.boundary +
+        " from " + startTime + " to " + endTime;
+
+      var dialog = new Dialog({
+        style: "width: 50%",
+        title: "Generate " + layerName + " report",
+        content: message,
+        actionBarTemplate: '' +
+          '<div class="dijitDialogPaneActionBar" data-dojo-attach-point="actionBarNode">' +
+          '  <button data-dojo-type="dijit/form/Button" type="submit" data-dojo-attach-point="closeButton">Cancel</button>' +
+          '  <button data-dojo-type="dijit/form/Button" type="button" data-dojo-attach-point="acceptButton">Accept</button>' +
+          '</div>'
+      });
+
+      $(dialog.closeButton).on("click", function() {
+        dialog.hide();
+      });
+
+      $(dialog.acceptButton).on("click", function() {
+        dialog.hide();
+
+        Logging.main.log(
+          "Visualization.UI.LayerReportDialog.report",
+          {
+            data: layerData,
+            toString: function() {
+              return this.data.boundary;
+            }
+          }
+        );
+
+        // TODO: Complete this once the server endpoint is done, eez should be
+        // replaced by the actual column name.
+        var url = layerData.reportBaseUrl + "/sub/eez=" + eez + "/report";
```

Comment:
@geowurster, the ticket is https://github.com/SkyTruth/pelagos-server/issues/810. Basically, what we want is, for a given EEZ, all the ships that were detected inside the EEZ in a period of time, with the date it was first detected in there and the date it was last seen in there.

What I want to know is where the EEZ field is in the new pipeline. I suppose that some task will classify each AIS record and append a field somewhere that indicates if the vessel was in a given EEZ. Is that right? And on the other hand, how do we identify a given EEZ? Do we have a numeric ID for each EEZ, some kind of code, or the name of the EEZ identifies it uniquely?